### ISSUE TITLE
`initialValues` dispatches two actions in deep forms

### ISSUE THREAD

**Issue by mohebifar** [erikras/redux-form#485]
I use server-side rendering in a project and in a form, there should be a `tickets[]` field with some array default value which comes from redux state. As a result of which, let's say the given array has two values, I must see two inputs with the following code:

```jsx
@reduxForm({
  form: 'testForm',
  fields: ['tickets[]', 'nonArrayValue']
}, state => ({ initialValues: state.folan.defaultTickets }))
class TestFormComponent extends Component {
  render() {
    return (<div>
      {this.props.fields.tickets.map(ticket => <input type="text" {...ticket} />)}
    </div>);
  }
}
```

It works well when there is no server-side rendering. But on the server, it seems that `tickets` array is empty. I logged the `tickets` value in the `render()` method then I checked the log on the browser's console and I realized that the component gets rendered two times. The first time, `tickets` is empty while other non-array values have a `defaultValue`. The second log shows that the `tickets` array has the correct value.

![Log preview](https://cloud.githubusercontent.com/assets/6104558/12050825/b7d0e6f4-af0f-11e5-818d-52d92e85e47e.png)

I also tried to defer rendering the component until an `INITIALIZE` action gets dispatched. It didn't work as well.

**Comment by erikras** [comment]
Is your `state.folan.defaultTickets` populated in the Redux store before you call `ReactDom.render()`?

**Comment by mohebifar** [comment]
Yes, it is.

**Comment by erikras** [comment]
Let me see if I'm understanding the flow:

1. `tickets` is empty on the server
2. Redux store is dehydrated and sent to the client
3. Redux store is rehydrated on the client
4. First render. (`tickets` still empty)
5. ??? Something fetches the data via AJAX ???
6. Second render (`tickets` populated)

If I'm wrong about 1, which I might be from your more recent comments, then you need to put some printouts in the code that is creating the Redux Store on the client to absolutely verify that the tickets data isn't getting lost between steps 2 and 4.

**Comment by mohebifar** [comment]
I created [this simple gist](https://gist.github.com/mohebifar/d22bd3bf123f57367c83). Maybe it clears out the issue.

**Comment by erikras** [comment]
Yes, that clears it up. It's amazing how much some sample code can clarify problems like this. Will investigate. :mag:

**Comment by erikras** [comment]
Hmm... So I wrote a test (see commit above) to check this problem, and it worked as expected. Can you see anything that is different?

**Comment by mohebifar** [comment]
That's it. Thanks :+1:

**Comment by shilpan** [comment]
I am facing this exact same issue. Has this been resolved?

**Comment by ooflorent** [comment]
Fixed in v5.1.4 (#843)
